services/brig/src/Brig/API/User.hs

```diff
@@ -239,23 +239,10 @@ createUser new = do
 
     pure (activatedTeam <|> joinedTeamInvite <|> joinedTeamSSO)
 
-  -- Handle e-mail activation (deprecated, see #RefRegistrationNoPreverification in /docs/reference/user/registration.md)
   edata <-
     if isJust teamInvitation
-      then return Nothing
-      else fmap join . for (userEmailKey <$> email) $ \ek -> case newUserEmailCode new of
-        Nothing -> do
-          timeout <- setActivationTimeout <$> view settings
-          edata <- lift $ Data.newActivation ek timeout (Just uid)
-          Log.info $
-            field "user" (toByteString uid)
-              . field "activation.key" (toByteString $ activationKey edata)
-              . msg (val "Created email activation key/code pair")
-          return $ Just edata
-        Just c -> do
-          ak <- liftIO $ Data.mkActivationKey ek
-          void $ activateWithCurrency (ActivateKey ak) c (Just uid) (join (bnuCurrency <$> newTeam)) !>> EmailActivationError
-          return Nothing
+      then handleEmailActivation email uid newTeam
```

Comment:
This seems like a behavior-changing change.  I've pushed 87e64c349665176931f9b9e3dc927e390150aab5, let's see what that does.  (This would also mean we have some test coverage issues.  Oh well, another day...)

Reply:
No I don't think so. Please see the https://github.com/wireapp/wire-server/pull/1670/commits/cab00b1640faf32b56f1d4cb343e008f78d295cd that introduced the refactoring.

Reply:
... Oh, yes, you were right. I made a mistake in the commit after that: https://github.com/wireapp/wire-server/pull/1670/commits/ed2accb4cdf4f400abd4599160b6beb8c4ff1ed4